server/sonar-server/src/main/java/org/sonar/server/component/ws/ShowAction.java

```diff
@@ -119,10 +118,9 @@ private static ShowWsResponse buildResponse(ComponentDto component, Organization
 
   private ComponentDto getComponentByUuidOrKey(DbSession dbSession, ShowWsRequest request) {
     ComponentDto component = componentFinder.getByUuidOrKey(dbSession, request.getId(), request.getKey(), ParamNames.ID_AND_KEY);
-    String projectUuid = firstNonNull(component.projectUuid(), component.uuid());
     if (!userSession.hasPermission(GlobalPermissions.SYSTEM_ADMIN) &&
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "hasPermission"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)